src/dredd.coffee

```diff
@@ -28,37 +28,85 @@ class Dredd
         duration: 0
     @configuration = applyConfiguration(config, @stats)
     configureReporters @configuration, @stats, @tests
-    @runner = new Runner(@configuration)
+    @runner = new Runner @configuration
 
   run: (callback) ->
     config = @configuration
     stats = @stats
 
-    fs.readFile config.blueprintPath, 'utf8', (loadingError, data) ->
-      return callback(loadingError, stats) if loadingError
-      reporterCount = config.emitter.listeners('start').length
-      config.emitter.emit 'start', data, () ->
-        reporterCount--
-        if reporterCount is 0
-          protagonist.parse data, blueprintParsingComplete
-
-    blueprintParsingComplete = (protagonistError, result) =>
-      return callback(protagonistError, config.reporter) if protagonistError
-      
-      if result['warnings'].length > 0
-        for warning in result['warnings']
-          message = 'Parser warning: ' + ' (' + warning.code + ') ' + warning.message
-          for loc in warning['location']
-            pos = loc.index + ':' + loc.length
-            message = message + ' ' + pos
-          logger.warn message
-      
-      runtime = blueprintAstToRuntime result['ast']
-      runtimeError = handleRuntimeProblems runtime
-      return callback(runtimeError, stats) if runtimeError
-
-      @runner.run runtime['transactions'], () =>
-        @transactionsComplete(callback)
+    config.files = []
+
+    # expand all globs
```

Comment:
Sorry! I should've described it better.

I just meant that I personally would find it easier to read (since it's a fairly long method) if it were split up into parts - and I thought the comments describing the parts suggested natural places to split.

A quick pseudo-example:

```coffeescript
expandAllGlobs(config)
removeDuplicateFilenames(config)
loadAllFiles(config)
parseAllBlueprints(config)
runAllTransactions(config)
```
Then each "task" would be self-contained in its own method/module/file/etc.